rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -271,145 +328,190 @@ void @(spec.base_type.pkg_name)_@(module_name)__destroy_ros_message(void * raw_r
 @[    end if]@
-  py@(field.name) = PyList_New(size@(field.name));
-  size_t idx@(field.name);
-  for (idx@(field.name) = 0; idx@(field.name) < size@(field.name); idx@(field.name)++) {
+    field = PyList_New(size);
+    if (!field) {
+      return NULL;
+    }
+    for (size_t i = 0; i < size; ++i) {
 @[    if field.type.type == 'char']@
-    PyList_SetItem(py@(field.name), idx@(field.name),
-      Py_BuildValue("C", tmpmessagedata@(field.name)[idx@(field.name)]));
+      int rc = PyList_SetItem(field, i, Py_BuildValue("C", src[i]));
+      assert(rc == 0);
 @[    elif field.type.type == 'byte']@
-    PyList_SetItem(py@(field.name), idx@(field.name),
-      PyBytes_FromStringAndSize((const char *)&tmpmessagedata@(field.name)[idx@(field.name)], 1));
+      int rc = PyList_SetItem(field, i, PyBytes_FromStringAndSize((const char *)&src[i], 1));
+      assert(rc == 0);
 @[    elif field.type.type == 'string']@
-    PyList_SetItem(py@(field.name), idx@(field.name), PyUnicode_FromString(tmpmessagedata@(field.name)[idx@(field.name)].data));
+      int rc = PyList_SetItem(field, i, PyUnicode_FromString(src[i].data));
+      assert(rc == 0);
 @[    elif field.type.type == 'bool']@
 @# using PyBool_FromLong because PyList_SetItem will steal ownership of the passed item
-    PyList_SetItem(py@(field.name), idx@(field.name), PyBool_FromLong(tmpmessagedata@(field.name)[idx@(field.name)] ? 1 : 0));
+      int rc = PyList_SetItem(field, i, PyBool_FromLong(src[i] ? 1 : 0));
+      assert(rc == 0);
 @[    elif field.type.type in ['float32', 'float64']]@
-    PyList_SetItem(py@(field.name), idx@(field.name), PyFloat_FromDouble(tmpmessagedata@(field.name)[idx@(field.name)]));
+      int rc = PyList_SetItem(field, i, PyFloat_FromDouble(src[i]));
+      assert(rc == 0);
 @[    elif field.type.type in [
         'int8',
         'int16',
         'int32',
     ]]@
-    PyList_SetItem(py@(field.name), idx@(field.name), PyLong_FromLong(tmpmessagedata@(field.name)[idx@(field.name)]));
+      int rc = PyList_SetItem(field, i, PyLong_FromLong(src[i]));
+      assert(rc == 0);
 @[    elif field.type.type in [
         'uint8',
         'uint16',
         'uint32',
     ]]@
-    PyList_SetItem(py@(field.name), idx@(field.name), PyLong_FromUnsignedLong(tmpmessagedata@(field.name)[idx@(field.name)]));
+      int rc = PyList_SetItem(field, i, PyLong_FromUnsignedLong(src[i]));
+      assert(rc == 0);
 @[    elif field.type.type == 'int64']@
-    PyList_SetItem(py@(field.name), idx@(field.name), PyLong_FromLongLong(tmpmessagedata@(field.name)[idx@(field.name)]));
+      int rc = PyList_SetItem(field, i, PyLong_FromLongLong(src[i]));
+      assert(rc == 0);
 @[    elif field.type.type == 'uint64']@
-    PyList_SetItem(py@(field.name), idx@(field.name), PyLong_FromUnsignedLongLong(tmpmessagedata@(field.name)[idx@(field.name)]));
+      int rc = PyList_SetItem(field, i, PyLong_FromUnsignedLongLong(src[i]));
+      assert(rc == 0);
```

Comment:
@dirk-thomas in release this gives unused variable warnings on osx and linux: http://ci.ros2.org/view/nightly/job/nightly_osx_release/632/

Reply:
Thanks for bringing this up. I will commit a fix-up later today.

Reply:
52830d1e45ad3f3cb470dfd3fd093735a205ead1 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=3366)](http://ci.ros2.org/job/ci_linux/3366/)